### ISSUE TITLE
add rule for none variable

### ISSUE THREAD

**Issue by tooptoop4** [astral-sh/ruff#8798]
below is not being flagged by any rule, i would expect it to suggest to flip the order of the conditions
```
try:
    return file_path.rsplit('/', 1)[1].startswith('redact') or not file_path
except Exception:
    return False
```

**Comment by charliermarsh** [comment]
Can you share a bit more on why you'd expect Ruff to suggest flipping the order?

**Comment by tmke8** [comment]
I assume it's because if `file_path` is `None`, then the evaluation of the first condition will throw a runtime error. But to catch this kind of mistake, ruff would need to know the type of `file_path`, so this seems more like a job for a static type checker.

**Comment by Skylion007** [comment]
This feels more under the scope of `type inference`? I think mypy can complain about this with the right settings.

**Comment by Avasam** [comment]
Yeah that's more of a type-checker thing.
Although even if it's not `None` it might be generally agreeable to check for truthyness first.

**Comment by charliermarsh** [comment]
I'm gonna close for now as this seems more an issue for a type checker. Thanks for filing!
